File: software/examples/php/ExampleCallback.php
<?php

require_once('Tinkerforge/IPConnection.php');
require_once('Tinkerforge/BrickletCompass.php');

use Tinkerforge\IPConnection;
use Tinkerforge\BrickletCompass;

const HOST = 'localhost';
const PORT = 4223;
const UID = 'XYZ'; // Change XYZ to the UID of your Compass Bricklet

// Callback function for heading callback
function cb_heading($heading)
{
    echo "Heading: " . $heading/10.0 . " °\n";
}

$ipcon = new IPConnection(); // Create IP connection
$c = new BrickletCompass(UID, $ipcon); // Create device object

$ipcon->connect(HOST, PORT); // Connect to brickd
// Don't use device before ipcon is connected

// Register heading callback to function cb_heading
$c->registerCallback(BrickletCompass::CALLBACK_HEADING, 'cb_heading');

// Set period for heading callback to 0.1s (100ms) without a threshold
$c->setHeadingCallbackConfiguration(100, FALSE, 'x', 0, 0);

echo "Press ctrl+c to exit\n";
$ipcon->dispatchCallbacks(-1); // Dispatch callbacks forever

?>

File: software/examples/perl/example_callback.pl
#!/usr/bin/perl

use strict;
use Tinkerforge::IPConnection;
use Tinkerforge::BrickletCompass;

use constant HOST => 'localhost';
use constant PORT => 4223;
use constant UID => 'XYZ'; # Change XYZ to the UID of your Compass Bricklet

# Callback subroutine for heading callback
sub cb_heading
{
    my ($heading) = @_;

    print "Heading: " . $heading/10.0 . " °\n";
}

my $ipcon = Tinkerforge::IPConnection->new(); # Create IP connection
my $c = Tinkerforge::BrickletCompass->new(&UID, $ipcon); # Create device object

$ipcon->connect(&HOST, &PORT); # Connect to brickd
# Don't use device before ipcon is connected

# Register heading callback to subroutine cb_heading
$c->register_callback($c->CALLBACK_HEADING, 'cb_heading');

# Set period for heading callback to 0.1s (100ms) without a threshold
$c->set_heading_callback_configuration(100, 0, 'x', 0, 0);

print "Press key to exit\n";
<STDIN>;
$ipcon->disconnect();
